library/public/types.h

```diff
@@ -206,6 +206,47 @@ class direction_t : public double_array_t<3>
   }
 };
 
+/**
+ *  Store a 3x3 transform matrix as a sequence of 9 float values
+ *  VTK expects a 4x4 matrix, but for 2d transforms this is exposed as a 3x3 matrix
+ */
+
+class transform2d_t : public double_array_t<9>
```

Comment:
I dont really see the point for a new type. a double_vector should work similarly afaict.

Reply:
I think it is needed. We could have specific constructors like `transform2d_t(double scaling[2], double translation[2], double angle)` with conversion to a matrix internally.

Reply:
https://github.com/f3d-app/f3d/issues/2301